modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +85,31 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = ::Proxy::Dns::Plugin.settings.dns_zones
```

Comment:
I'd prefer if this was initialized in the constructor; and an accessor was used instead of referring to ```::Proxy::Dns::Plugin.settings``` directly. 

This property is a required one, you should validate that it's present in dns_dnscmd_plugin.rb: ```validate_presence :dns_zones```. I'll update the validator to handle arrays (right now validation will pass even if the array is empty).